lightning/src/ln/channelmanager.rs

```diff
@@ -6928,11 +6928,10 @@ where
 		&self, prev_hop: HTLCPreviousHopData, payment_preimage: PaymentPreimage,
 		payment_info: Option<PaymentClaimDetails>, completion_action: ComplFunc,
 	) {
-		let counterparty_node_id =
-			match self.short_to_chan_info.read().unwrap().get(&prev_hop.short_channel_id) {
-				Some((cp_id, _dup_chan_id)) => Some(cp_id.clone()),
-				None => None
-			};
+		let counterparty_node_id = prev_hop.counterparty_node_id.or_else(|| {
+			let short_to_chan_info = self.short_to_chan_info.read().unwrap();
+			short_to_chan_info.get(&prev_hop.short_channel_id).map(|(cp_id, _)| *cp_id)
```

Comment:
nit: drop the local variable for a single, chained expression

Reply:
YMMV, but personally I'd find it more readable at a glance than when having `short_to_chan_info` repeated three times.

Reply:
Right, I don't mind that. More so the density of the code.